lib/blockTemplate.js

```diff
@@ -131,9 +137,11 @@ var BlockTemplate = module.exports = function BlockTemplate(
 
         if (this.txCount <= 0x7f){
             var varInt = new Buffer(txCount, 'hex');
-        }
-        else if (this.txCount <= 0x7fff){
-            var varInt = new Buffer.concat([Buffer('FD', 'hex'), new Buffer(txCount, 'hex')]);
+        } else if (this.txCount <= 0x7fff) {
+            if (txCount.length == 2) {
```

Comment:
This is a good fix, initially from https://github.com/z-classic/node-stratum-pool/issues/39